sdk-extensions/incubator/src/test/java/io/opentelemetry/sdk/extension/incubator/fileconfig/ConfigurationReaderTest.java

```diff
@@ -0,0 +1,335 @@
+import io.opentelemetry.sdk.extension.incubator.fileconfig.internal.model.Console;
+import io.opentelemetry.sdk.extension.incubator.fileconfig.internal.model.ExplicitBucketHistogram;
+import io.opentelemetry.sdk.extension.incubator.fileconfig.internal.model.Headers;
+import io.opentelemetry.sdk.extension.incubator.fileconfig.internal.model.LogRecordExporter;
+import io.opentelemetry.sdk.extension.incubator.fileconfig.internal.model.LogRecordLimits;
+import io.opentelemetry.sdk.extension.incubator.fileconfig.internal.model.LogRecordProcessor;
+import io.opentelemetry.sdk.extension.incubator.fileconfig.internal.model.LoggerProvider;
+import io.opentelemetry.sdk.extension.incubator.fileconfig.internal.model.MeterProvider;
+import io.opentelemetry.sdk.extension.incubator.fileconfig.internal.model.MetricExporter;
+import io.opentelemetry.sdk.extension.incubator.fileconfig.internal.model.MetricReader;
+import io.opentelemetry.sdk.extension.incubator.fileconfig.internal.model.OpentelemetryConfiguration;
+import io.opentelemetry.sdk.extension.incubator.fileconfig.internal.model.Otlp;
+import io.opentelemetry.sdk.extension.incubator.fileconfig.internal.model.OtlpMetric;
+import io.opentelemetry.sdk.extension.incubator.fileconfig.internal.model.ParentBased;
+import io.opentelemetry.sdk.extension.incubator.fileconfig.internal.model.PeriodicMetricReader;
+import io.opentelemetry.sdk.extension.incubator.fileconfig.internal.model.Prometheus;
+import io.opentelemetry.sdk.extension.incubator.fileconfig.internal.model.PullMetricReader;
+import io.opentelemetry.sdk.extension.incubator.fileconfig.internal.model.Resource;
+import io.opentelemetry.sdk.extension.incubator.fileconfig.internal.model.Sampler;
+import io.opentelemetry.sdk.extension.incubator.fileconfig.internal.model.Selector;
+import io.opentelemetry.sdk.extension.incubator.fileconfig.internal.model.SimpleSpanProcessor;
+import io.opentelemetry.sdk.extension.incubator.fileconfig.internal.model.SpanExporter;
+import io.opentelemetry.sdk.extension.incubator.fileconfig.internal.model.SpanLimits;
+import io.opentelemetry.sdk.extension.incubator.fileconfig.internal.model.SpanProcessor;
+import io.opentelemetry.sdk.extension.incubator.fileconfig.internal.model.Stream;
+import io.opentelemetry.sdk.extension.incubator.fileconfig.internal.model.TraceIdRatioBased;
+import io.opentelemetry.sdk.extension.incubator.fileconfig.internal.model.TracerProvider;
+import io.opentelemetry.sdk.extension.incubator.fileconfig.internal.model.View;
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+import org.junit.jupiter.api.Test;
+
+@SuppressWarnings("unchecked")
+class ConfigurationReaderTest {
+
+  @Test
+  void read_KitchenSinkExampleFile() throws IOException {
+    OpentelemetryConfiguration.OpentelemetryConfigurationBuilder expectedBuilder =
```

Comment:
is it possible to make the builder a top-level class?

Reply:
The alternative to inner class builders is chained setters as described in here: https://github.com/joelittlejohn/jsonschema2pojo/issues/506

I think it reads pretty well and is fairly ergonomic. I don't think copy on write memory hit will be a big deal in this situation.